src/pretix/base/logentrytypes.py

```diff
@@ -0,0 +1,252 @@
+    """
+    Base class for a type of LogEntry, identified by its action_type.
+    """
+
+    def __init__(self, action_type=None, plain=None):
+        if action_type:
+            self.action_type = action_type
+        if plain:
+            self.plain = plain
+
+    def display(self, logentry):
+        """
+        Returns the message to be displayed for a given logentry of this type.
+
+        :return: `str` or `LazyI18nString`
+        """
+        if hasattr(self, 'plain'):
+            plain = str(self.plain)
+            if '{' in plain:
+                data = defaultdict(lambda: '?', logentry.parsed_data)
+                return plain.format_map(data)
+            else:
+                return plain
+
+    def get_object_link_info(self, logentry) -> dict:
+        """
+        Return information to generate a link to the `content_object` of a given log entry.
+
+        Not implemented in the base class, causing the object link to be omitted.
+
+        :return: Dictionary with the keys ``href`` (containing a URL to view/edit the object) and ``val`` (containing the
+                 escaped text for the anchor element)
+        """
+        pass
+
+    def get_object_link(self, logentry):
+        a_map = self.get_object_link_info(logentry)
+        return make_link(a_map, self.object_link_wrapper)
+
+    object_link_wrapper = '{val}'
+
+    def shred_pii(self, logentry):
+        """
+        To be used for shredding personally identified information contained in the data field of a LogEntry of this
+        type.
+        """
+        raise NotImplementedError
+
+
+class EventLogEntryType(LogEntryType):
+    """
+    Base class for any `LogEntry` type whose `content_object` is either an `Event` itself or belongs to a specific `Event`.
+    """
+
+    def get_object_link_info(self, logentry) -> dict:
+        if hasattr(self, 'object_link_viewname') and hasattr(self, 'object_link_argname') and logentry.content_object:
+            return {
+                'href': reverse(self.object_link_viewname, kwargs={
+                    'event': logentry.event.slug,
+                    'organizer': logentry.event.organizer.slug,
+                    self.object_link_argname: self.object_link_argvalue(logentry.content_object),
+                }),
+                'val': escape(self.object_link_display_name(logentry.content_object)),
+            }
+
+    def object_link_argvalue(self, content_object):
+        """Return the identifier used in a link to content_object."""
+        return content_object.id
+
+    def object_link_display_name(self, content_object):
+        """Return the display name to refer to content_object in the user interface."""
+        return str(content_object)
+
+
+class OrderLogEntryType(EventLogEntryType):
+    object_link_wrapper = _('Order {val}')
+    object_link_viewname = 'control:event.order'
+    object_link_argname = 'code'
+
+    def object_link_argvalue(self, order):
+        return order.code
```

Comment:
What if we need more than one additional argument for the URL? Probably currently not the case anywhere, but wouldn't it be more flexible to have a `object_link_args`, which returns a dict that gets merged with the other URL-args?

```suggestion

    def object_link_args(self, order):
        return {'code': order.code}
```

Maybe for the default case, where the `logentry.content_object` is the only additional URL-param, we could keep the `object_link_argname` and change the original `object_link_argvalue` to

```
    def object_link_args(self, content_object):
        """Return the identifier used in a link to content_object."""
        if self.object_link_argname:
            return {self.object_link_argname: content_object.id}
        return {}
```